src/Esfa.Vacancy.Register.Infrastructure/Repositories/VacancyRepository.cs

```diff
@@ -1,13 +1,74 @@
-﻿using System.Threading.Tasks;
+﻿using Dapper;
+using Esfa.Vacancy.Register.Domain.Repositories;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using DomainEntities = Esfa.Vacancy.Register.Domain.Entities;
 
 namespace Esfa.Vacancy.Register.Infrastructure.Repositories
 {
     public class VacancyRepository : IVacancyRepository
     {
-        public async Task<Domain.Entities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
+        private readonly IProvideSettings _provideSettings;
+
+        public VacancyRepository(IProvideSettings provideSettings)
+        {
+            _provideSettings = provideSettings;
+        }
+
+        public async Task<DomainEntities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
         {
-            await Task.Delay(10);
-            return null;
+            var connectionString =
+                _provideSettings.GetSetting(ApplicationSettingConstants.AvmsPlusDatabaseConnectionStringKey);
+
+            DomainEntities.Vacancy vacancy;
+
+            using (var sqlConn = new SqlConnection(connectionString))
+            {
+                await sqlConn.OpenAsync();
+                var results =
+                    await sqlConn.QueryAsync<DomainEntities.Vacancy>(
+                        GetVacancyDetailsQuery, new { ReferenceNumber = referenceNumber });
+                vacancy = results.FirstOrDefault();
+            }
+
+            return vacancy;
         }
+
+        const string GetVacancyDetailsQuery = @"
+SELECT  V.VacancyReferenceNumber AS Reference
+,       V.Title
+,       V.ShortDescription
+,       V.[Description]
+,       V.VacancyTypeId
+,       V.WageUnitId
+,       V.WeeklyWage
+,       V.WorkingWeek
+,       V.WageText
+,       V.HoursPerWeek
+,       V.DurationValue AS ExpectedDuration
+,       V.ExpectedStartDate
+,		VH.HistoryDate AS DatePosted
+,       V.ApplicationClosingDate
+,       V.NumberofPositions 
+,       V.EmployerDescription
+,       V.EmployersWebsite
+,       V.TrainingTypeId
+,       RS.LarsCode AS LarsStandardId
+,       CAST(AO.ShortName AS INT) AS SSA1Code
+FROM[dbo].[Vacancy]        V
+WITH(NOLOCK)
```

Comment:
Is there an issue with uncommitted transactions in the database? 

[When should you use "with (nolock)"](https://stackoverflow.com/questions/686724/when-should-you-use-with-nolock)
> SQL Server 2005 fixed most of the bugs that made NOLOCK necessary. So unless you are using SQL Server 2000 or > earlier, you shouldn't need it.